### DISCUSSION TITLE
What causes rust analyzer to spawn `cargo test --no-run` ?

### DISCUSSION THREAD

**Discussion by augustocdias** [rust-lang/rust-analyzer#18096] [Q&A] [answered]
So since a couple weeks I can barely use rust-analyzer in neovim. It got to a point that my machine went out of memory and restarted itself. After investigating I found out that dozens(sometimes hundreds) of cargo test processes are automatically spawned... I have no idea what is causing this... Could it be rust-analyzer? How do I troubleshoot this and find the root cause? I have tried a minimal config with just the `rustaceannvim` plugin but I couldn't reproduce... So the plugin is not likely the cause, but I don't know what else to look at. Could you give me some help on how to find the issue?

This is the command that is ran:

```
cargo test --no-run --package core-crypto --lib --message-format=json
```

**Comment by Veykril** [comment]
Nothing in rust-analyzer itself passes `--no-run` to cargo (only the vscode extension does, but that obviously doesn't apply to you with neovim here)

**Comment by Veykril** [comment, marked as answer]
The problem likely lies with `rustaceannvim` given https://github.com/mrcjkb/rustaceanvim/blame/1997658e2865f06b0dd67cf824ee8e71d347c185/lua/rustaceanvim/overrides.lua#L76-L93

**Reply by augustocdias** [reply]
Yeah. I've just found out it is causing this behavior when checking the server status and configuring dap. Thanks for the help

**Reply by augustocdias** [reply]
The author claimed that the command is issued by rust analyzer. That it just request analyzer for the runnables and analyzer issues the cargo commands.

**Reply by Veykril** [reply]
Then I don't know where, the server source has no occurrences of `--no-run` when I grepped it

**Reply by augustocdias** [reply]
You're right... I dug deeper into the code of the plugin and it requests the runnables from the server, with the results it calls `cargo test --no-run` for each result in order to configure the debug adapter.
